src/web/app/pages-instructor/student-list/__snapshots__/student-list.component.spec.ts.snap

```diff
@@ -0,0 +1,3044 @@
+        <tr
+          s=""
+        >
+          <td
+            class="align-center"
+          >
+            
+            
+            <a
+              href="null"
+              role="button"
+            >
+              View Photo
+            </a>
+            
+          </td>
+          
+          <td>
+            Tutorial Group 1
+          </td>
+          <td>
+            Team 1
+          </td>
+          <td>
+            Emma Farrell
+          </td>
+          <td
+            class="align-center"
+          >
+            Joined
+          </td>
+          <td>
+            emma.f.tmms@gmail.tmt
+          </td>
+          <td
+            class="no-print align-center"
+          >
+            
+            
+            <a
+              class="btn btn-light btn-sm mr-2"
+              href="/web/instructor/courses/student/details?courseid=&studentemail=emma.f.tmms@gmail.tmt"
+              ng-reflect-klass="btn btn-light btn-sm mr-2"
+              ng-reflect-ng-class="[object Object]"
+              ng-reflect-ngb-tooltip="View the details of the studen"
+              ng-reflect-query-params="[object Object]"
+              ng-reflect-router-link="/web/instructor/courses/studen"
+            >
+               View
+            </a>
+            
+            
+            <a
+              class="btn btn-light btn-sm mr-2"
+              href="/web/instructor/courses/student/edit?courseid=&studentemail=emma.f.tmms@gmail.tmt"
+              ng-reflect-klass="btn btn-light btn-sm mr-2"
+              ng-reflect-ng-class="[object Object]"
+              ng-reflect-ngb-tooltip="Use this to edit the details o"
+              ng-reflect-query-params="[object Object]"
+              ng-reflect-router-link="/web/instructor/courses/studen"
+            >
+               Edit
+            </a>
+            
+            
+            
+            <button
+              class="btn btn-light btn-sm mr-2"
+              ng-reflect-klass="btn btn-light btn-sm mr-2"
+              ng-reflect-ng-class="[object Object]"
+              ng-reflect-ngb-tooltip="Delete the student and the cor"
+            >
+              Delete
+            </button>
+            
+            
+            <a
+              class="btn btn-light btn-sm mr-2"
+              href="/web/instructor/students/records?courseid=&studentemail=emma.f.tmms@gmail.tmt"
+              ng-reflect-klass="btn btn-light btn-sm mr-2"
+              ng-reflect-ng-class="[object Object]"
+              ng-reflect-ngb-tooltip="View all data about this stude"
+              ng-reflect-query-params="[object Object]"
+              ng-reflect-router-link="/web/instructor/students/recor"
+            >
+               All Records
+            </a>
+            
+          </td>
+        </tr>
+        
+      </tbody>
+    </table>
+  </div>
+</tm-student-list>
+`;
+
+exports[`StudentListComponent should snap with some student list data and some students to hide 1`] = `
```

Comment:
Next time, make life easier for reviewers by giving the hidden students names like 'This should not be displayed' :(

Reply:
Well, I am not writing this test to make life easier for reviewers. I have fed in data that you are more likely to see in production for this application. I have preferably tried to make it as real as possible (so we are more likely to catch bugs).

As a reviewer, you could have looked out for:
```
    component.listOfStudentsToHide = [
      'alice.b.tmms@gmail.tmt',
      'tester@tester.com',
    ];
```

From there you would have to make sure that the student with this email, is not going to be present in this snapshot. 

It is definitely a little bit more work and I agree that names like 'This should not be displayed' would have made things a little faster (its also not super critical to have names like Alice).  I can change this if you prefer but personally, I would stick to data that is as close to production if I could. 

If you follow a pattern like this, you may end up compromising the quality of your tests at some point.

Reply:
Just basing it off #9605 - I haven't traditionally follow it either, but I kinda see the merit in it. No worries, this is fine, just for future reference anyway :)